### ISSUE TITLE
Redstone detector

### ISSUE THREAD

**Issue by aodzip** [ForgeEssentials/ForgeEssentialsMain#1223]
Any way to detect activied RedStone?

**Comment by luacs1998** [comment]
No, but if you want to consider it a feature request then it can be done.

**Comment by olee** [comment]
How do you mean that?

**Comment by aodzip** [comment]
@olee use command to find activited redstone . For some player will make too many clocks to lag the whole server

**Comment by olee** [comment]
Hm that would mean we would need to monitor redstone activations for let's say 1 second and report all locations, where more than a certain threshold of activations ocurred.
I will look into whether and how we can track redstone signals.

**Comment by luacs1998** [comment]
Hmm... 

Can it just be a special testfor that detects the redstone output level at a given coordinate?

**Comment by olee** [comment]
With testfor you would get all active redstone, but that would be a lot of activity in some regions (e.g. just an active lever+redstone+lamp and it would not tell the admin about redstone circuits that make servers lag etc.
That's why if we want to solve this, it would need to be done properly like the solution I described.

**Comment by luacs1998** [comment]
I was thinking something simple like "here's a coordinate, tell me if its emitting redstone, and at what level".

**Comment by olee** [comment]
@aodzip If I understood your request correctly, you want to be able to especially detect things like redstone-timers and other active stuff like that, correct?

**Comment by aodzip** [comment]
Right.
There is a easy way
just list out the location of redstone things which are loaded

从我的 ASUS 发送

-------- 原邮件 --------
发件人:Björn Zeutzheim <notifications@github.com>
已发送:Thu, 15 Jan 2015 01:13:52 +0800
收件人:ForgeEssentials/ForgeEssentialsMain <ForgeEssentialsMain@noreply.github.com>
抄送:aodzip <aodzip@gmail.com>
主题:Re: [ForgeEssentialsMain] Redstone detector (#1223)

**Comment by olee** [comment]
You mean all redstone in the vicinity?
That would be very much in some situations and displaying that information to the user would become difficult then...
Detecting only redstion activity (not just state = active) would allow a much more precise detection...

**Comment by aodzip** [comment]
There is a bukkit plugin c called RedstoneClockDetector use /rcd command to find these things
poor English sorry

从我的 ASUS 发送

**Comment by olee** [comment]
Your english is not bad at all, cirno :wink: 
And the most important thing: You properly explain your problems :smile: 
I will take a look at RCD.

**Comment by AnrDaemon** [comment]
Probably you don't need "all active redstone", but "redstone state changes".
And if some chunk emit more changes than other, it may be up for investigation.
